linalg/slepc.hpp

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef MFEM_SLEPC
+#define MFEM_SLEPC
+
+#include "../config/config.hpp"
+
+#ifdef MFEM_USE_SLEPC
+#ifdef MFEM_USE_MPI
+
+#include "petsc.hpp"
+
+// Forward declarations
+typedef struct _p_EPS *EPS;
+
+namespace mfem
+{
+
+void MFEMInitializeSlepc();
+void MFEMInitializeSlepc(int*,char***);
+void MFEMInitializeSlepc(int*,char***,const char[],const char[]);
+void MFEMFinalizeSlepc();
+
+class SlepcEigenSolver
+{
+private:
+   /// Boolean to handle SetFromOptions calls
+   mutable bool clcustom;
+
+   /// SLEPc linear eigensolver object
+   EPS eps;
+   /// Solver tolerance
+   double _tol;
+
+   /// Maximum number of iterations
+   int _max_its;
+
+   /// Real and imaginary part of eigenvector
+   mutable PetscParVector *VR, *VC;
+
+   /// Boolean to handle SetOperator calls
+   mutable bool operatorset;
+
+public:
+   /// Constructors
+   SlepcEigenSolver(MPI_Comm comm, const std::string &prefix = std::string());
+
+   virtual ~SlepcEigenSolver();
+
+   /// Set solver tolerance
+   void SetTol(double tol);
+
```

Comment:
You may want to add typecasting to get the EPS object (see the PETSc classes) of the SlepcEigensolver

Reply:
Good point.

Reply:
Isn't what https://github.com/mfem/mfem/blob/45881cbdd5744d3878635803d3a020c8f79223b5/linalg/slepc.hpp#L112 does? Or you also want typecasting to PetscObject?

Reply:
Sorry, missed that. Thanks
Since you mentioned, also add the PetscObject conversion to match what is done in petsc.hpp for  the other exposed PETSc classes

Reply:
I will add that.